crates/wash-lib/src/start/nats.rs

```diff
@@ -271,31 +275,69 @@ where
             config.port
         );
     }
-    let child = if let Some(parent_path) = bin_path.as_ref().parent() {
+
+    let bin_path_ref = bin_path.as_ref();
+
+    if let Some(parent_path) = bin_path_ref.parent() {
         let config_path = parent_path.join(NATS_SERVER_CONF);
         let host = config.host.clone();
         let port = config.port;
-        config.write_to_path(&config_path).await?;
-        Command::new(bin_path.as_ref())
-            .stderr(stderr)
+
+        let mut cmd_args = vec![
+            "-js".to_string(),
+            "--addr".to_string(),
+            host,
+            "--port".to_string(),
+            port.to_string(),
+            "--config".to_string(),
+        ];
+
+        if let Some(nats_cfg_path) = &config.config_path {
+            if nats_cfg_path.is_dir() {
```

Comment:
Consider using [`anyhow::ensure`](https://docs.rs/anyhow/latest/anyhow/macro.ensure.html) here!

Something like:

```rust
anyhow::ensure!(nats_cfg_path.is_dir(), "The provided config...")
```

With that, we can remove one `if` branch here and make it easier to read